packages/@lwc/ssr-runtime/src/index.ts

```diff
@@ -210,74 +181,116 @@ export class LightningElement implements PropsAvailableAtConstruction {
     }
-    getElementsByTagName(_qualifiedName: unknown): HTMLCollectionOf<Element> {
-        throw new Error('Method "getElementsByTagName" not implemented.');
+    get lastChild(): never {
+        throw new TypeError('"getLastChild" is not supported in this environment');
     }
-    hasAttribute(_qualifiedName: string): boolean {
-        throw new Error('Method "hasAttribute" not implemented.');
+    get lastElementChild(): never {
+        throw new TypeError('"getLastElementChild" is not supported in this environment');
     }
-    hasAttributeNS(_namespace: string | null, _localName: string): boolean {
-        throw new Error('Method "hasAttributeNS" not implemented.');
+    get ownerDocument(): never {
+        // Intentionally not "get*" to match @lwc/engine-server
+        throw new TypeError('"ownerDocument" is not supported in this environment');
     }
-    querySelector(_selectors: string): Element | null {
-        throw new Error('Method "querySelector" not implemented.');
+    get style(): never {
+        // Intentionally not "get*" to match @lwc/engine-server
+        throw new TypeError('"style" is not supported in this environment');
+    }
+
+    attachInternals(): never {
+        throw new TypeError('"attachInternals" is not supported in this environment');
     }
-    querySelectorAll(_selectors: string): NodeListOf<Element> {
-        throw new Error('Method "querySelectorAll" not implemented.');
+    dispatchEvent(_event: Event): never {
+        throw new TypeError('"dispatchEvent" is not supported in this environment');
+    }
+    getBoundingClientRect(): never {
+        throw new TypeError('"getBoundingClientRect" is not supported in this environment');
+    }
+    getElementsByClassName(_classNames: string): never {
+        throw new TypeError('"getElementsByClassName" is not supported in this environment');
+    }
+    getElementsByTagName(_qualifiedName: unknown): never {
+        throw new TypeError('"getElementsByTagName" is not supported in this environment');
+    }
+    querySelector(_selectors: string): never {
+        throw new TypeError('"querySelector" is not supported in this environment');
+    }
+    querySelectorAll(_selectors: string): never {
+        throw new TypeError('"querySelectorAll" is not supported in this environment');
+    }
+
+    // -------------------------------------------------------------------------------- //
+    // Stubs to satisfy the HTMLElementTheGoodParts (from @lwc/engine-core) interface //
+    // The interface is not explicitly referenced here, so this may become outdated //
+    // -------------------------------------------------------------------------- //
+
+    accessKey?: string;
+    dir?: string;
+    draggable?: boolean;
+    hidden?: boolean;
+    id?: string;
+    lang?: string;
+    shadowRoot?: ShadowRoot | null;
+    spellcheck?: boolean;
+    tabIndex?: number;
+    title?: string;
```

Comment:
Aside from `shadowRoot`, these are all global attributes. Should they be implemented as getters/setters that just reflect onto `__attrs`?

Reply:
Yes, we'll need to implement proper property reflection. This is already handled in `@lwc/engine-server` because it gets it from `base-lightning-element.ts` in `@lwc/engine-core`. We should also add tests for it if they don't exist.